### PR TITLE
QUIC: Rename SSL_tick, SSL_get_tick_timeout

### LINKED ISSUE #20867: QUIC: rename SSL_tick()
Rename `SSL_tick()` to `SSL_process_events()` or `SSL_handle_events()`.  Also rename `SSL_get_tick_timeout()` appropriately.  Finally, fix all the documentation.

### PR DESCRIPTION
**Pull request by hlandau**
Built on top of MSMT PR.

```
9a3f93ddfb QUIC: Update documentation to reflect compatibility of DTLS APIs
12e12b4ec4 QUIC APL: Allow DTLSv1 APIs to be used for compatibility
3d1142a376 QUIC: Update documentation for SSL_get_event_timeout
5dfa5ae60e QUIC APL: Change SSL_get_event_timeout API design
b4976d91bb QUIC: Rename SSL_tick, SSL_get_tick_timeout
8ec61607a8 QUIC Documentation: Rename SSL_tick, SSL_get_tick_timeout
```

Fixes #20867, #20868.

### CODE REVIEW

**Comment by mattcaswell** [comment]
Needs updating how that #20580 has merged.

**Comment by hlandau** [comment]
Built on MSST so will wait until that's merged.

**Comment by hlandau** [comment]
Sorry, typo - this is built on MSMT.

**Comment by hlandau** [comment]
Rebased and ready for review.

File: doc/man3/SSL_get_event_timeout.pod
Comment (mattcaswell) [inline doc/man3/SSL_get_event_timeout.pod:5]: Rephrase "needs to be ticked", since we're not using the tick terminology any more.

File: doc/man3/SSL_get_event_timeout.pod
Comment (mattcaswell) [inline doc/man3/SSL_get_event_timeout.pod:28]: "set to zero" is slightly confusing since `tv` points to a struct type not an integer type.

File: doc/man3/SSL_get_rpoll_descriptor.pod
Comment (mattcaswell) [inline doc/man3/SSL_get_rpoll_descriptor.pod:48]: Should be enclosed in `L<>`

File: ssl/quic/quic_impl.c
Comment (mattcaswell) [inline ssl/quic/quic_impl.c:839]: perhaps rephrase by avoid the word "ticking"

File: ssl/quic/quic_impl.c
Comment (mattcaswell) [inline ssl/quic/quic_impl.c:856]: ticked?

**Review by mattcaswell — CHANGES_REQUESTED** [review]
Needs yet another rebase!

**Comment by hlandau** [comment]
Updated.

File: doc/man3/DTLSv1_handle_timeout.pod
Comment (mattcaswell) [inline doc/man3/DTLSv1_handle_timeout.pod:25]: typo: this this

File: doc/man3/SSL_handle_events.pod
Comment (mattcaswell) [inline doc/man3/SSL_handle_events.pod:41]: ticking?

File: doc/man3/SSL_handle_events.pod
Comment (mattcaswell) [inline doc/man3/SSL_handle_events.pod:43]: ticking?

**Review by mattcaswell — CHANGES_REQUESTED** [review]

**Comment by hlandau** [comment]
Updated.

**Review by mattcaswell — APPROVED** [review]

**Comment by mattcaswell** [comment]
@t8m please reconfirm after the latest change

**Review by t8m — APPROVED** [review]

**Comment by t-j-h** [comment]
It wouldn't hurt in all the test programs to check the return code from SSL_handle_events but approved with or without that haven't been dealt with.

**Comment by t8m** [comment]
> It wouldn't hurt in all the test programs to check the return code from SSL_handle_events but approved with or without that haven't been dealt with.

That could be a separate PR

**Comment by t8m** [comment]
Merged to master branch. Thank you!

**Pull request closed without merging**
